File: name_parsers/parse_name.py
def parse_name(name):
    parsed_name = {
        "has_prefix": False,
        "prefix": "",
        "has_suffix": False,
        "suffix": "",
        "plus_one": False,
        "name": ""
    }

    if name[0] == '"':
        parsed_name["has_prefix"] = True
        split_name = name.split('"')
        parsed_name["prefix"] = split_name[1]
        name = split_name[2][1:]  # NOTE - We're trimming the leading whitespace. We're modifying our `name` arg

    if " of " in name:
        parsed_name["has_suffix"] = True
        split_name = name.split(" of ")
        parsed_name["suffix"] = "of " + split_name[1]
        # Check for +1
        if " +1" in parsed_name["suffix"]:
            parsed_name["plus_one"] = True
            parsed_name["suffix"] = parsed_name["suffix"].replace(" +1", "")
        name = split_name[0]  # NOTE - We're modifying our `name` arg

    parsed_name["name"] = name
    return parsed_name

File: name_parsers/parseName.js
let parseName = (name) => {
  let parsedName = {
    hasPrefix: false,
    prefix: "",
    hasSuffix: false,
    suffix: "",
    plusOne: false,
    name: ""
  }

  let splitName;

  // Check for prefix
  if(name[0] === "\"") {
    parsedName.hasPrefix = true;
    splitName = name.split("\"");
    parsedName.prefix = splitName[1];
    name = splitName[2];  // NOTE - we modify the function arg `name`
  }

  // Check for suffix
  if(name.indexOf(" of ") >= 0) {
    parsedName.hasSuffix = true;
    splitName = name.split(" of ");
    parsedName.suffix = "of " + splitName[1];
    if(parsedName.suffix.indexOf(" +1") >= 0) {
      parsedName.plusOne = true;
      parsedName.suffix = parsedName.suffix.split(" +1")[0];
    }
    name = splitName[0];  // NOTE - we modify the function arg `name`
  }

  parsedName.name = name.trim();

  return parsedName;
}
